### PR TITLE
support the new cryptography 1.2 method of getting key type

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

File: src/OpenSSL/crypto.py [hunk 1]
```diff
@@ -248,7 +248,12 @@ def check(self):
         if self._only_public:
             raise TypeError("public key only")
 
-        if _lib.EVP_PKEY_type(self._pkey.type) != _lib.EVP_PKEY_RSA:
+        if hasattr(self._pkey, "type"):
+            key_type = self._pkey.type
+        else:
+            key_type = _lib.Cryptography_EVP_PKEY_id(self._pkey)
+
+        if _lib.EVP_PKEY_type(key_type) != _lib.EVP_PKEY_RSA:
```
Comment (alex) [inline src/OpenSSL/crypto.py]: This can just be `self.type()`?

Comment (reaperhulk) [inline src/OpenSSL/crypto.py]: Yup, wasn't sure if we should refactor as part of this though. I can do that though.

Comment (alex) [inline src/OpenSSL/crypto.py]: Yes, I think it's a clear improvement.

**Comment by alex** [comment]
Coverage miss is because we don't have CI for an older version of cryptography. I think this is ok though.

**Comment by hynek** [comment]
We must add a tox/travis target for the last cryptography that worked the old way so we don’t break it in the future.

**Comment by reaperhulk** [comment]
I can put a try catch back in here so that we don't have to raise the cryptography minimum version (which will prevent the pypy version requirement bump due to cffi 1.0+), but someone else will need to add a CI target that uses an older cryptography to get coverage.

**Comment by hynek** [comment]
proceed please :)

**Comment by hynek** [comment]
fixed in #418

**Pull request closed without merging**